hash.c

```diff
@@ -6528,68 +6539,235 @@ env_update(VALUE env, VALUE hash)
 }
 
 /*
- *  A Hash is a dictionary-like collection of unique keys and their values.
- *  Also called associative arrays, they are similar to Arrays, but where an
- *  Array uses integers as its index, a Hash allows you to use any object
- *  type.
+ *  A \Hash is a dictionary-like collection of key-value pairs
```

Comment:
"A Hash maps unique keys to specific values." ?

I'm not sure why referring to a dictionary is a good idea except that the naming in general is kind of confusing.

- Python calls it a `dict` (dictionary). Which is normally word-definition pairs.
- Ruby calls it a `Hash` but typically "hash" means a digest of a value.

The most logical name IMHO is a "Map". Internally, Ruby uses this in some other places `ObjectSpace::WeakMap` for example. However, we have to make do with the name as given.

Reply:
Fixed, with clarification that a key has only one value.